README.md

```diff
@@ -1,69 +1,320 @@
+
+    // Algo delay in milliseconds.
+    //
+    const ALGO_DELAY_MS: u16 = 500;
+
+    loop {
+        // Reset the buffer at the start of each iteration
+        //
+        buf_writer.reset();
+
+        // Get SPL value from the sensor.
+        //
+        let spl = pa_spl.get_latest_decibel().unwrap();
+
+        // Format string with SPL value, then covert to string.
+        //
+        write!(buf_writer, "SPL: {}\r", spl).unwrap();
+        let spl_str = buf_writer.as_str();
+
+        // Write the string out to the UART.
+        //
+        uart4.write_str(spl_str).unwrap_or_else(|_| {
+            loop {
+                // Failed to write to UART4.
+                asm::nop(); // If real app, replace with actual error handling.
+            }
+        });
+
+        // Limit algorithm to (1000 * (1 / UART_WRITE_DELAY_MS)) Hz.
+        delay.delay_ms(ALGO_DELAY_MS);
+    }
 }
+
 ```
 
-See the [example project](https://github.com/EngJay/pa-spl/blob/main/examples/read-decibel-value/README.md) for
-a complete example for a STM32F3 Discovery.
+## Tests
 
-## Features
+Two test suites are included - an off-target unit test suite included in the
+source of the driver that uses
+[embedded-hal-mock](https://github.com/dbrgn/embedded-hal-mock) and a
+hardware-in-the-loop (HIL) test suite that is included as a nested cargo project
+following the [pattern](https://ferrous-systems.com/blog/test-driver-crate/)
+described by Ferrous Systems for testing a driver crate.
+
+To run the off-target tests:
+
+```cli
+cargo test
+```
+
+The
+[HIL testing project](https://github.com/EngJay/pa-spl/tree/main/target-tests/README.md)
+is in the `target-tests` directory and is configured to use
+[probe-rs](https://probe.rs/) to automatically build the tests, flash the
+target, run the tests, and report results.
+
+To run the HIL tests, connect the hardware according to the
+[STM32F3 Discovery example](https://github.com/EngJay/pa-spl/blob/main/examples/read-decibel-value/README.md)
+provided in the repo (minus the TTL-USB convertor) with a spectrum
+analysis version of the PCB Artists sensor, 0x32 or 0x33, then:
+
+```cli
+cd target-tests
+cargo test
+```
+
+The CI configuration runs the tests for every commit and PR against Rust 1.65,
+stable, beta, and nightly. The HIL tests are run privately due to the security
+issues with attaching a local GitHub Actions runner to a public respository but
+anyone is welcome to run them locally on their own hardware.
+
+## Minimum Supported Rust Version (MSRV)
 
-- Read SPL value.
-- Set window for averaging of SPL value over time.
-- Set thresholds for triggering of interrupt pin that can be used as an external
-  trigger.
+This crate is guaranteed to compile on stable Rust
+[1.65](https://releases.rs/docs/1.60.0/) and up. It might compile with older
+versions but that may change in any new patch release.
 
-## Rust Version Support
+See [here](https://github.com/EngJay/pa-spl/tree/main/docs/msrv.md) for details
```

Comment:
Hm, this is an interesting issue because the README is included in the docs, which are automatically hosted on https://docs.rs/, so the relative path will result in a broken link there, yet including the full path will result in the wrong version of the document loading. I wonder if this can be pinned by using the tag in the URL? BRB.

Reply:
The links have been updated to preemptively point to the 0.1.0 tag, so they will work off GitHub, outside the repo, and point to the correct version of the files. Might be a better way to handle this but this will get me through the initial release.